vng_api_common/models.py

```diff
@@ -69,106 +61,3 @@ class Meta:
 
     def __str__(self):
         return self.identifier
-
-
-class APICredential(models.Model):
-    """
-    Store credentials for external APIs.
-
-    When we need to authenticate against a remote API, we need to know which
-    client ID and secret to use to sign the JWT.
-    """
-
-    api_root = models.URLField(
-        _("API-root"),
-        unique=True,
-        help_text=_(
-            "URL of the external API, ending in a trailing slash. Example: https://example.com/api/v1/"
-        ),
-    )
-    label = models.CharField(
-        _("label"),
-        max_length=100,
-        default="",
-        help_text=_("Human readable label of the external API."),
-    )
-    client_id = models.CharField(
-        _("client ID"),
-        max_length=255,
-        help_text=_("Client ID to identify this API at the external API."),
-    )
-    secret = models.CharField(
-        _("secret"), max_length=255, help_text=_("Secret belonging to the client ID.")
-    )
-    user_id = models.CharField(
-        _("user ID"),
-        max_length=255,
-        help_text=_(
-            "User ID to use for the audit trail. Although these external API credentials are typically used by"
-            "this API itself instead of a user, the user ID is required."
-        ),
-    )
-    user_representation = models.CharField(
-        _("user representation"),
-        max_length=255,
-        default="",
-        help_text=_("Human readable representation of the user."),
-    )
-
-    class Meta:
-        verbose_name = _("external API credential")
-        verbose_name_plural = _("external API credentials")
-
-    def __str__(self):
-        return self.api_root
-
-    @classmethod
-    def get_auth(cls, url: str, **kwargs) -> Union[ClientAuth, None]:
-        split_url = urlsplit(url)
-        scheme_and_domain = urlunsplit(split_url[:2] + ("", "", ""))
-
-        candidates = (
-            cls.objects.filter(api_root__startswith=scheme_and_domain)
-            .annotate(api_root_length=Length("api_root"))
-            .order_by("-api_root_length")
-        )
-
-        # select the one matching
-        for candidate in candidates.iterator():
-            if url.startswith(candidate.api_root):
-                credentials = candidate
-                break
-        else:
-            return None
-
-        auth = ClientAuth(
-            client_id=credentials.client_id,
-            secret=credentials.secret,
-            user_id=credentials.user_id,
-            user_representation=credentials.user_representation,
-            **kwargs,
-        )
-        return auth
-
-
-class ClientConfig(SingletonModel):
```

Comment:
Why is this abstract model deleted? I'm pretty sure it's used in Open Zaak. Maybe just change api_root to be a FK to `Service` model instead of the url